src/routes.js

```diff
@@ -17,39 +18,64 @@ function routes () {
         UserService = locator.get('UserService');
 
   app.get('/', function (req, res) {
-    let params = {};
+
     let pageIndex = req.query.page || 1;
     let offset = (pageIndex * 10) - 10;
 
-    params.sequelize = {
-      include: [{ model: Episode, include: [Podcast] }],
-      where: isClipMediaRefWithTitle,
-      offset: offset
-    };
-
-    params.paginate = {
-      default: 10,
-      max: 200
-    };
-
-    return ClipService.find(params)
-      .then(page => {
-
-        // TODO: handle 404 if beyond range of page object
-
-        let total = page.total;
-        let showNextButton = offset + 10 < total ? true : false;
-        let clips = page.data;
-        res.render('home/index.html', {
-          clips: clips,
-          pageIndex: pageIndex,
-          showNextButton: showNextButton,
-          currentPage: 'Home Page'
-        });
-      })
-      .catch(e => {
-        console.log(e);
+    let queryObj = {
+      metrics: 'ga:uniquePageviews',
+      dimensions: 'ga:pagePath',
+      startDate: '7daysAgo',
+      endDate: 'today',
+      sort: '-ga:uniquePageviews',
+      // maxResults: only 10 will load on a page, but we don't want
+      // clips without titles to appear in the list
+      maxResults: 30,
+      startIndex: offset || 1,
+      filters: 'ga:pagePath=~/clips'
+    }
+
+    return new Promise((resolve, reject) => {
+      queryGoogleApiData(resolve, reject, queryObj)
+    })
+    .then(data => {
```

Comment:
That sounds important. I've never implemented caching and don't know what it looks like. Is there a library or guide you would recommend?